src/deep-db/lib/DynamoDB/EventualConsistency.js

```diff
@@ -0,0 +1,346 @@
+   * @param {Kernel} deepKernel
+   * @param {Object} modelsToExtend
+   */
+  constructor(dynamoDb, dynamoDbDocumentClient, deepKernel, modelsToExtend) {
+    this._dynamoDb = dynamoDb;
+    this._dynamoDbDocumentClient = dynamoDbDocumentClient;
+    this._kernel = deepKernel;
+    this._modelsToExtend = modelsToExtend;
+    this._localModel = false;
+    this._sqsInstances = {};
+  }
+
+  /**
+   * @returns {Object}
+   */
+  get _queuesModelMapping() {
+    return this._kernel.config.dbOffloadQueues || {};
+  }
+
+  /**
+   * @returns {AWS.SQS|*}
+   */
+  _sqs(modelName) {
+    if (!this._sqsInstances.hasOwnProperty(modelName)) {
+      let queuesMapping = this._queuesModelMapping;
+
+      if (!queuesMapping.hasOwnProperty(modelName)) {
+        return null;
+      }
+
+      this._sqsInstances[modelName] = new AWS.SQS({
+        region: EventualConsistency.getRegionFromSqsQueueUrl(queuesMapping[modelName].url),
+      });
+    }
+
+    return this._sqsInstances[modelName];
+  }
+
+  /**
+   * @param {String} modelName
+   * @param {String} method
+   * @param {Object} payload
+   * @param {Function} callback
+   * @private
+   */
+  _sendPayload(modelName, method, payload, callback) {
+    let queuesMapping = this._queuesModelMapping;
+
+    if (!queuesMapping.hasOwnProperty(modelName)) {
+      return callback(null);
+    }
+
+    let params = {
+      MessageBody: JSON.stringify({method, payload,}),
+      QueueUrl: queuesMapping[modelName].url,
+    };
+
+    this._sqs(modelName).sendMessage(params, error => callback(error));
+  }
+
+  /**
+   * @param {String} queueUrl
+   * @returns {String}
+   */
+  static getRegionFromSqsQueueUrl(queueUrl) {
+    let regionParts = queueUrl.match(/\.([^\.]+)\.amazonaws\.com\/.*/i);
+
+    if (!regionParts || regionParts.length === 0) {
+      throw new Error(queueUrl, 'Unable to extract AWS region.');
+    }
+
+    return regionParts[1];
+  }
+
+  /**
+   * @param {Boolean} state
+   * @returns {EventualConsistency}
+   */
+  localMode(state) {
+    this._localModel = !!state;
+
+    return this;
+  }
+
+  /**
+   * @returns {AWS.DynamoDB|AWS.DynamoDB.DocumentClient|*}
+   */
+  extend() {
+
+    // avois doing it for older deploys
+    if (this._kernel.config.dbOffloadQueues) {
+      EventualConsistency.PROXIED_METHODS.forEach((method) => {
+        let originalMethod = `__deep_db_EC_${method}__`;
+
+        this._dynamoDbDocumentClient[originalMethod] = this._dynamoDbDocumentClient[method];
+        this._dynamoDbDocumentClient[method] = this._ecExecCb(method, originalMethod);
+      });
+
+      this._extendModels();
+      this._dynamoDbDocumentClient[EventualConsistency.DEEP_DB_DECORATOR_FLAG] = true;
+    }
+
+    return this._dynamoDbDocumentClient;
+  }
+
+  /**
+   * @private
+   */
+  _extendModels() {
+    let queuesMapping = this._queuesModelMapping;
+    let tablesNames = this._kernel.config.tablesNames;
+
+    for (let modelName in this._modelsToExtend) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.](https://www.codacy.com/app/MitocGroup/deep-framework/file/2724293608/issues/source?bid=2895475&fileBranchId=3348235#l126)